test/fixtures/repo-commit-template/dot-git/hooks/fsmonitor-watchman.sample

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use IPC::Open2;
+
+# An example hook script to integrate Watchman
```

Comment:
can you help me understand more about what this shell script is doing?  bash isn't my strong suit, and I don't understand how this relates to the rest of your pull request.